src/main/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/domain/handlers/presubmit/RequestHomeOfficeDataHandler.java

```diff
@@ -0,0 +1,249 @@
+import uk.gov.hmcts.reform.iahomeofficeintegrationapi.domain.entities.ApplicationStatus;
+import uk.gov.hmcts.reform.iahomeofficeintegrationapi.domain.entities.AsylumCase;
+import uk.gov.hmcts.reform.iahomeofficeintegrationapi.domain.entities.CodeWithDescription;
+import uk.gov.hmcts.reform.iahomeofficeintegrationapi.domain.entities.HomeOfficeCaseStatus;
+import uk.gov.hmcts.reform.iahomeofficeintegrationapi.domain.entities.HomeOfficeMetadata;
+import uk.gov.hmcts.reform.iahomeofficeintegrationapi.domain.entities.HomeOfficeSearchResponse;
+import uk.gov.hmcts.reform.iahomeofficeintegrationapi.domain.entities.Person;
+import uk.gov.hmcts.reform.iahomeofficeintegrationapi.domain.entities.RejectionReason;
+import uk.gov.hmcts.reform.iahomeofficeintegrationapi.domain.entities.ccd.DynamicList;
+import uk.gov.hmcts.reform.iahomeofficeintegrationapi.domain.entities.ccd.callback.Callback;
+import uk.gov.hmcts.reform.iahomeofficeintegrationapi.domain.entities.ccd.callback.PreSubmitCallbackResponse;
+import uk.gov.hmcts.reform.iahomeofficeintegrationapi.domain.entities.ccd.callback.PreSubmitCallbackStage;
+import uk.gov.hmcts.reform.iahomeofficeintegrationapi.domain.handlers.PreSubmitCallbackHandler;
+import uk.gov.hmcts.reform.iahomeofficeintegrationapi.domain.service.FeatureToggler;
+import uk.gov.hmcts.reform.iahomeofficeintegrationapi.infrastructure.client.util.HomeOfficeDateFormatter;
+
+@Slf4j
+@Component
+public class RequestHomeOfficeDataHandler implements PreSubmitCallbackHandler<AsylumCase> {
+
+    private static final String PROBLEM_MESSAGE = "### There is a problem\n\n";
+
+    private static final String HOME_OFFICE_CALL_ERROR_MESSAGE = PROBLEM_MESSAGE
+            + "The service has been unable t"
+            + "o retrieve the Home Office information about this appeal.\n\n"
+            + "[Request the Home Office information](/case/IA/Asylum/${[CASE_REFERENCE]}/"
+            + "trigger/requestHomeOfficeData) to try again. This may take a few minutes.";
+
+    private final FeatureToggler featureToggler;
+
+    private HomeOfficeDataErrorsHelper homeOfficeDataErrorsHelper;
+
+    public RequestHomeOfficeDataHandler(HomeOfficeDataErrorsHelper homeOfficeDataErrorsHelper,
+                                        FeatureToggler featureToggler) {
+
+        this.homeOfficeDataErrorsHelper = homeOfficeDataErrorsHelper;
+        this.featureToggler = featureToggler;
+    }
+
+    @Override
+    public boolean canHandle(
+            PreSubmitCallbackStage callbackStage,
+            Callback<AsylumCase> callback
+    ) {
+        requireNonNull(callbackStage, "callbackStage must not be null");
+        requireNonNull(callback, "callback must not be null");
+
+        return callbackStage == PreSubmitCallbackStage.ABOUT_TO_SUBMIT
+                && callback.getEvent() == REQUEST_HOME_OFFICE_DATA
+                && featureToggler.getValue("home-office-uan-feature", false);
```

Comment:
Ideally home-office-uan-feature needs to be checked in case-api itself and the home-office-integration-api should not need to introduce home-office-uan-feature check.